client/db/bolt/testdata/blueprint.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+// This file should compiled from the commit the file was introduced, otherwise
+// it may not compile due to API changes, or may not create the database with
+// the correct old version.  This file should not be updated for API changes.
```

Comment:
I'm afraid I don't understand this comment.  "This file should compiled from the commit the file was introduced, otherwise it may not compile due to API changes" says to me not to ever update this file and if a commit updates the file I should go back to the initial commit for when the file was created. And I don't know what to make of 'This file should not be updated for API changes."  Please clarify.

Reply:
The file will generate the testdata needed for verifying it associated test, it is likely for API changes to occur while this file has already been used to generate the testdata needed for the test. The comments there is saying if the testdata needs to be regenerate for some reason revert to the commit the file was introduced to regenerate the testdata instead of updating it due to API changes.

Reply:
Will stand by for your feedback on this before I push the next commit resolving issues raised for this PR.

Reply:
Oh, could it just say "always regenerate test data with this function prior to testing" or "ensure to generate test data with the same revision of this function as the revision of the code being tested"?

> The comments there is saying if the testdata needs to be regenerate for some reason revert to the commit the file was introduced to regenerate the testdata instead of updating it due to API changes.

I still don't get the above.  Are we really saying to checkout the code commit where this file was first created to generate the test data?

Reply:
The testdata is expected to be  generated and committed with the file used to generate it. `blueprint.go` doesn't have a corresponding test data file because it is only showing the process and not actually generating any data to test against. An actual upgrade would commit the testdata as well, take a look [here](https://github.com/decred/dcrpool/commit/adb290d94374d91c597ad0b17bff142e8bacd483#diff-0efd2bec14b740a80d891a70c8f57162)

Reply:
OK, that makes more sense now.  (1) blueprint.go is a template for what to do for a future upgrade, and (2) the test data would get checked in.  Seeing https://github.com/decred/dcrdex/pull/460 helped make that clear.